applications/views.py

```diff
@@ -127,11 +127,15 @@ def page(request, pk, key):
     except ValueError:
         raise Http404
 
+    unbound_form = page.get_unbound_data_form()
+
     if request.method == "GET":
-        form = page.get_unbound_data_form()
+        form = unbound_form
```

Comment:
Not sure we need this optimisation any more!

Reply:
:eyes: I got blinded to what I'd changed from what was originally present; thanks! Just edited the commit.